guide/src/class.md

```diff
@@ -337,10 +337,27 @@ impl SubSubClass {
         let super_ = self_.into_super(); // Get PyRef<'_, SubClass>
         SubClass::method2(super_).map(|x| x * v)
     }
+
+    #[staticmethod]
+    fn dynamic_type(py: Python<'_>, val: usize) -> PyResult<PyObject> {
```

Comment:
What about naming this something like...
```suggestion
    fn factory_method(py: Python<'_>, val: usize) -> PyResult<PyObject> {
```
...because I think that when people want something like this they'll search for the "factory" pattern.